File: frontend-api/src/main/java/no/nav/melosys/tjenester/gui/dto/trygdeavgift/AvgiftsperiodeDto.java
package no.nav.melosys.tjenester.gui.dto.trygdeavgift;

import java.math.BigDecimal;
import java.time.LocalDate;

import no.nav.melosys.domain.avgift.Avgiftsperiode;
import no.nav.melosys.domain.kodeverk.Trygdedekninger;

public class AvgiftsperiodeDto {
    private final LocalDate fom;
    private final LocalDate tom;
    private final Trygdedekninger trygdedekning;
    private final BigDecimal avgiftssats;
    private final BigDecimal avgiftPerMd;

    public AvgiftsperiodeDto(LocalDate fom,
                             LocalDate tom,
                             Trygdedekninger trygdedekning,
                             BigDecimal avgiftssats,
                             BigDecimal avgiftPerMd) {
        this.fom = fom;
        this.tom = tom;
        this.trygdedekning = trygdedekning;
        this.avgiftssats = avgiftssats;
        this.avgiftPerMd = avgiftPerMd;
    }

    public LocalDate getFom() {
        return fom;
    }

    public LocalDate getTom() {
        return tom;
    }

    public Trygdedekninger getTrygdedekning() {
        return trygdedekning;
    }

    public BigDecimal getAvgiftssats() {
        return avgiftssats;
    }

    public BigDecimal getAvgiftPerMd() {
        return avgiftPerMd;
    }

    static AvgiftsperiodeDto av(Avgiftsperiode avgiftsperiode) {
        return new AvgiftsperiodeDto(
            avgiftsperiode.getFom(),
            avgiftsperiode.getTom(),
            avgiftsperiode.getTrygdedekning(),
            avgiftsperiode.getAvgiftssats(),
            avgiftsperiode.getAvgiftPerMd()
        );
    }
}

File: integrasjon/src/main/kotlin/no/nav/melosys/integrasjon/dokgen/dto/innvilgelseftrl/AvgiftsperiodeDto.kt
package no.nav.melosys.integrasjon.dokgen.dto.innvilgelseftrl

import tools.jackson.databind.annotation.JsonSerialize
import tools.jackson.databind.ext.javatime.ser.LocalDateSerializer
import no.nav.melosys.domain.kodeverk.Inntektskildetype
import java.math.BigDecimal
import java.time.LocalDate

data class AvgiftsperiodeDto(
    @JsonSerialize(using = LocalDateSerializer::class) val fom: LocalDate,
    @JsonSerialize(using = LocalDateSerializer::class) val tom: LocalDate,
    val avgiftssats: BigDecimal?,
    val avgiftPerMd: BigDecimal,
    val inntektskildetype: Inntektskildetype,
    val avgiftspliktigInntektPerMd: BigDecimal,
    val beregningsregel: String? = null,
)
